hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/ConsumerSenderConfiguration.java

```diff
@@ -265,7 +265,8 @@ public FutureAsyncTimeout futureAsyncTimeoutFactory(InstrumentedExecutorServiceF
         ScheduledExecutorService timeoutExecutorService = executorFactory.getScheduledExecutorService(
                 "async-timeout",
                 senderAsyncTimeoutProperties.getThreadPoolSize(),
-                senderAsyncTimeoutProperties.isThreadPoolMonitoringEnabled()
+                senderAsyncTimeoutProperties.isThreadPoolMonitoringEnabled(),
+                false
```

Comment:
What do You think about splitting `InstrumentedExecutorServiceFactory.getScheduledExecutorService()` to two methods (overloading it), one with boolean parameter `removeOnCancel`. Thanks to this we would not have misleading hardcoding parameters and changes in consumers.

Reply:
Sure, I have created a builder which will be more future-proof, please take a look